packages/block-library/src/navigation/editor.scss

```
@@ -20,6 +20,15 @@
 	}
 }
 
+// The following rule affects the positioning of the dropdown arrow indicator.
+// On the frontend, this element is inline, which makes it look correct. In the editor,
+// the label is block, which causes the dropdown indicator to wrap onto a new line.
+// Therefore we set it to inline-block.
+.wp-block-navigation-item__label {
+	display: inline-block;
```

The `inline-block` prevents the inheritance of the text decoration style. I've tested with setting it to `inline` only and across multiple themes this now allows the text decoration to be inherited as well as maintaining the indicator's correct position.

```suggestion
	display: inline;
```